whisper_medusa/models/model.py

```diff
@@ -170,11 +225,28 @@ def __init__(self, config):
                         )
                     )
                 self.medusa_heads.append(nn.Sequential(*head_list))
+        elif self.config.medusa_heads_type == "medusa_block":
+            self.medusa_block = WhisperDecoderLayer(self.whisper_model.config)
+            self.medusa_block.load_state_dict(
+                self.whisper_model.model.decoder.layers[-1].state_dict()
+            )  # load the last layer of the whisper model
+            self.medusa_heads = nn.ModuleList()
+            for _ in range(self.config.medusa_num_heads):
+                head_list = []
+                for _ in range(self.config.medusa_num_layers):
+                    head_list.append(
+                        MedusaResBlock(
+                            self.config.d_model, self.config.medusa_hidden_size
+                        )
+                    )
+                self.medusa_heads.append(nn.Sequential(*head_list))   
```

Comment:
encapsulate to private func, same for base_head case and consider using dict to avoid lots of if else (now it is ok since we have only two variants).

Reply:
done